### PR TITLE
Only allocate IP addresses for neutron on compute nodes

### PR DESCRIPTION
**Pull request by vuntz**
We were doing giving the controller an IP on os_sdn (which makes no
sense for the nova controller).

Also, vmware compute nodes don't need direct access to networking.

### CODE REVIEW

**Comment by rossella** [comment]
+1

**Comment by tboerger** [comment]
LGTM

File: crowbar_framework/app/models/nova_service.rb [hunk 1]
```diff
@@ -184,7 +184,18 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
 
     neutron = ProposalObject.find_proposal("neutron",role.default_attributes["nova"]["neutron_instance"])
 
-    all_nodes.each do |n|
+    compute_nodes_for_network = []
+    role.override_attributes["nova"]["elements"].each do |role, nodes|
+      # only care about compute nodes
+      next unless role =~ /^nova-multi-compute-/
+      # vmware compute nodes do not need access to the networking
+      next if role == "nova-multi-compute-vmware"
```
Comment (rhafer) [inline crowbar_framework/app/models/nova_service.rb]: What about hyperv nodes? They don't need any IP on the sdn network either, do they?

Comment (vuntz) [inline crowbar_framework/app/models/nova_service.rb]: Why wouldn't they need it? Or let me rephrase: they don't really need it because they only support vlan for now, but I don't think we should keep this restriction in the long term and instead use ovs for windows.

The rationale I used is really: is the compute node just a proxy to something else, or is it actually hosting stuff?

Comment (rhafer) [inline crowbar_framework/app/models/nova_service.rb]: > Why wouldn't they need it? 

Honestly I am not even sure if IP allocations for something else than the admin network even work with the current state of the hyperv recipes. But otoh they shouldn't really hurt either. So I don't have any objection against merging this.
